src/mbgl/storage/file_source_request.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <mbgl/storage/file_source.hpp>
+#include <mbgl/util/async_request.hpp>
+
+#include <memory>
+#include <functional>
+
+namespace mbgl {
+
+class Mailbox;
+
+class FileSourceRequest : public AsyncRequest {
+public:
+    FileSourceRequest(std::shared_ptr<Mailbox> mailbox);
+    ~FileSourceRequest() final;
+
+    void onResponse(FileSource::Callback&& callback);
+    void onCancel(std::function<void()>&& callback);
+    void setResponse(const Response& res);
+
+private:
+    void runResponseCallback();
+
+    std::shared_ptr<Mailbox> mailbox;
```

Comment:
We're only storing this to make sure it continues to live as long as the request lives? If the Mailbox lived as long as the surrounding `*FileSource`, we could get rid of this pointer.